### PR TITLE
[Contributors Polygon] Updated UI to allow contributors add polygons as well.

### LINKED ISSUE #228: [Layer dialog] "Contributors may add" dropdown
image

Default to only points.

### PR DESCRIPTION
**Pull request by HemanParbhakar**
Fixes #228 

This PR add checkbox options allowing contributors to select points and polygons for a Layer.

@gino-m @DaoyuT

### CODE REVIEW

**Comment by HemanParbhakar** [comment]
Screenshot 2021-09-02 at 2 05 28 PM

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 1]
```diff
@@ -133,7 +134,11 @@ export class LayerDialogComponent implements OnDestroy {
     const canAddPoints = this.layer?.contributorsCanAdd?.find(
       val => val === 'points'
     );
-    this.contributorsCanAdd = canAddPoints ? true : false;
+    const canAddPolygons = this.layer?.contributorsCanAdd?.find(
+      val => val === 'polygons'
+    );
+    this.contributorsCanAddPoint = canAddPoints ? true : false;
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts:140]: => `this.contributorsCanAddPoint = this.layer?.contributorsCanAdd?.includes('points')`

Same for Polygons. You can then delete the above calls to `find`

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 2]
```diff
@@ -57,7 +57,8 @@ export class LayerDialogComponent implements OnDestroy {
   form?: Form;
   @ViewChildren(FormFieldEditorComponent)
   formFieldEditors?: QueryList<FormFieldEditorComponent>;
-  contributorsCanAdd = true;
+  contributorsCanAddPoint = true;
+  contributorsCanAddPolygon = true;
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts:61]: Point => Points
Polygon => Polygons

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 3]
```diff
@@ -163,14 +168,22 @@ export class LayerDialogComponent implements OnDestroy {
     const fields = this.layerService.convertFieldsListToMap(this.fields);
     const formId = this.form?.id;
     const forms = this.layerService.createForm(formId, fields);
+    let contributions: string[] = [];
+    if (this.contributorsCanAddPoint && this.contributorsCanAddPolygon) {
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts:172]: if (this.contributorsCanAddPoints) {
  allowedFeatureTypes.push('points');
}

Same for polygons.

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 4]
```diff
@@ -163,14 +168,22 @@ export class LayerDialogComponent implements OnDestroy {
     const fields = this.layerService.convertFieldsListToMap(this.fields);
     const formId = this.form?.id;
     const forms = this.layerService.createForm(formId, fields);
+    let contributions: string[] = [];
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts:171]: s/contributions/allowedFeatureTypes/

**Review by gino-m — CHANGES_REQUESTED** [review]

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 1 above)
Comment (HemanParbhakar) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: updated

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 3 above)
Comment (HemanParbhakar) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: updated

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 2 above)
Comment (HemanParbhakar) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: updated

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 4 above)
Comment (HemanParbhakar) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: updated

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts [hunk 5]
```diff
@@ -173,13 +169,14 @@ export class LayerDialogComponent implements OnDestroy {
...
     const forms = this.layerService.createForm(formId, fields);
-    let contributions: string[] = [];
-    if (this.contributorsCanAddPoint && this.contributorsCanAddPolygon) {
-      contributions = ['points', 'polygons'];
-    } else if (this.contributorsCanAddPolygon) {
-      contributions = ['polygons'];
-    } else if (this.contributorsCanAddPoint) {
-      contributions = ['points'];
+    const allowedFeatureTypes: string[] = [];
+    if (this.contributorsCanAddPoints && this.contributorsCanAddPolygons) {
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.ts:173]: These conditional blocks are overlapping/redundant - please see code in previous comment for solution.

**Review by gino-m — CHANGES_REQUESTED** [review]

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.ts (hunk 5 above)
Comment (HemanParbhakar) [reply web-ng/src/app/components/layer-dialog/layer-dialog.component.ts]: updated.

**Review by gino-m — APPROVED** [review]

**Pull request merged**